tls-codec/src/tls_vec.rs

```diff
@@ -0,0 +1,225 @@
+                    }
+                    fn visit_seq<V>(self, mut seq: V) -> Result<$name<T>, V::Error>
+                    where
+                        V: serde::de::SeqAccess<'de>,
+                    {
+                        let vec = seq
+                            .next_element()?
+                            .ok_or_else(|| serde::de::Error::invalid_length(0, &self))?;
+                        Ok($name::<T>::new(vec))
+                    }
+                    fn visit_map<V>(self, mut map: V) -> Result<$name<T>, V::Error>
+                    where
+                        V: serde::de::MapAccess<'de>,
+                    {
+                        let mut vec = None;
+                        while let Some(key) = map.next_key()? {
+                            match key {
+                                Field::Vec => {
+                                    if vec.is_some() {
+                                        return Err(serde::de::Error::duplicate_field("vec"));
+                                    }
+                                    vec = Some(map.next_value()?);
+                                }
+                            }
+                        }
+                        let vec = vec.ok_or_else(|| serde::de::Error::missing_field("vec"))?;
+                        Ok($name::<T>::new(vec))
+                    }
+                }
+                deserializer.deserialize_struct(
+                    stringify!($name),
+                    &["vec"],
+                    TlsVecVisitor {
+                        data: std::marker::PhantomData,
+                    },
+                )
+            }
+        }
+
+        impl<T: Serialize + Deserialize + Clone + PartialEq> Serialize for $name<T> {
+            fn tls_serialize(&self, buffer: &mut Vec<u8>) -> Result<(), Error> {
+                let len = self.vec.len();
+                if len > (<$size>::MAX as usize) {
+                    return Err(Error::InvalidVectorLength);
+                }
+                (self.vec.len() as $size).tls_serialize(buffer)?;
+                for e in self.vec.iter() {
+                    e.tls_serialize(buffer)?;
+                }
+                Ok(())
+            }
+        }
+
+        impl<T: Serialize + Deserialize + Clone + PartialEq> Deserialize for $name<T> {
+            fn tls_deserialize(cursor: &Cursor) -> Result<Self, Error> {
+                let mut result = Self { vec: Vec::new() };
+                let len = <$size>::tls_deserialize(cursor)?;
+                let sub_cursor = cursor.sub_cursor(len.try_into()?)?;
+                while sub_cursor.has_more() {
+                    result.push(T::tls_deserialize(&sub_cursor)?);
+                }
+                Ok(result)
+            }
+        }
+    };
+}
+
+impl_tls_vec!(u8, TlsVecU8);
+impl_tls_vec!(u16, TlsVecU16);
+impl_tls_vec!(u32, TlsVecU32);
+impl_tls_vec!(u64, TlsVecU64);
```

Comment:
See the comment about `usize` earlier. The MLS spec doesn't use 64 bit vectors afaik. Maybe dropping this would just resolve the problem.

Reply:
Yeah I drop it here with a reference to #319